lib/core/application/payment/add_new_card_bloc/add_new_card_bloc.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:app/core/domain/payment/payment_repository.dart';
+import 'package:bloc/bloc.dart';
+import 'package:flutter_stripe/flutter_stripe.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'add_new_card_state.dart';
+
+part 'add_new_card_bloc.freezed.dart';
+
+class AddNewCardBloc extends Cubit<AddNewCardState> {
+  AddNewCardBloc(this._paymentRepository) : super(AddNewCardState.initial());
+
+  final PaymentRepository _paymentRepository;
+
+  void onCardHolderNameChange(String? value) {
+    emit(state.copyWith(cardHolderName: value));
+  }
+
+  void onCardNumberChange(String? value) {
+    emit(state.copyWith(cardNumber: value));
+  }
+
+  void onValidDateChange(String? value) {
+    emit(state.copyWith(validThrough: value));
+  }
+
+  void onCvvChange(String? value) {
+    emit(state.copyWith(cvv: value));
+  }
+
+  Future<void> addNewCard() async {
+    emit(state.copyWith(status: AddNewCardBlocStatus.loading));
+    // For debugging purpose
+    Stripe.publishableKey = 'pk_test_TYooMQauvdEDq54NiTphI7jx';
```

Comment:
please define this in env file and AppConfig

Reply:
should  the key assignment should happen in some init function ?

Reply:
This is only for testing purpose, when integrate with BE we will have publishableKey elsewhere